contracts/rust/adapter/src/sol_types.rs

```diff
@@ -230,6 +259,42 @@ impl<'de> Deserialize<'de> for ValidatorRegistered {
     }
 }
 
+impl Serialize for ValidatorRegisteredV2 {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
why are we adding on this? so `.abi_encode()` is not sufficient?

Reply:
We store the events locally in the DB for that I think they are wrapped into an enum. 

https://github.com/EspressoSystems/espresso-network/blob/7617c2204f25ae9dec89f6418ee8429133a31cbc/types/src/v0/v0_3/stake_table.rs#L96-L105

It would  probably also work with abi_encode / decode but to change now we could  have to migrate storage of any nodes that have this deployed already. We would still have to implement serde I think to work with our storage code.

Correct me if I'm wrong @imabdulbasit .